### PR TITLE
Upgrade framework and example to Swift 4.2

### PR DESCRIPTION
**Pull request by a2**
### Checklist
- [x] I've tested my changes.
- [x] I've read the [Contribution Guidelines](https://github.com/alexaubry/BulletinBoard/blob/master/CONTRIBUTING.md).
- [x] I've updated the documentation if necessary.

### Motivation and Context
Issue #97.

### Description
Upgrade the Swift version of the framework and sample project from Swift 3 to Swift 4.2.

### CODE REVIEW

File: BulletinBoard.podspec [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name         = "BulletinBoard"
-  s.version      = "2.0.1"
+  s.version      = "3.0.0"
```
Comment (AndrewBarba) [inline BulletinBoard.podspec]: Can you add `s.swift_version = "4.2"`? This was added so CP can officially set the correct `SWIFT_VERSION` for the target

**Comment by danthorpe** [comment]
Hey @a2 @alexaubry - just wondering what the status of this is? It looks like version got bumped to 3.0.0 and then again (but back) to 2.0.2?

Not sure if it's related, but I'm seeing some odd compile errors _for a UI Test target_ when using Cocoapods 1.6.0.beta.1, Xcode 10, Mojave.

**Comment by stevenp** [comment]
Since Swift 3 mode is deprecated in Xcode 10.1, is there any chance of getting this released soon?

**Comment by danthorpe** [comment]
> @danthorpe Could you open an issue with your problems with the UI test target compilation so I can have a look?

@alexaubry sorry, turned out to be some other Xcode & Cocoapods configuration issue, unrelated to BB.

**Comment by arvidurs** [comment]
Is there a timeframe to implement this? Its a major dependency in my framework

File: BulletinBoard.podspec (hunk 1 above)
Comment (alexaubry) [reply BulletinBoard.podspec]: Added it, thanks for the heads-up 👍

**Pull request merged**